src/test/ui/editions/auxiliary/edition-kw-macro-2015.rs

```diff
@@ -4,25 +4,35 @@
 
 #[macro_export]
 macro_rules! produces_async {
-    () => (pub fn async() {})
+    () => {
+        pub fn async() {}
+    };
```

Comment:
This is in the wrong direction. The error output looks like `async` is on its own line and I wanted `pub fn async` to not be split into multiple lines.

Is it possible the error reporting itself is doing something weird here, and printing `r#async` without it actually being a code snippet?

Reply:
I didn't immediately figure out what eddyb meant in this commit, but it appears to be this diff:
```diff
-LL | () => (pub fn r#async () { })
+LL | r#async
```
Error reporting machinery no longer fetches the whole `r#async`'s line for displaying, only the `r#async` itself.

Reply:
So I tried:
```rust

        let rmeta::MacroDef { body, legacy } = data.get_macro(id.index, sess);
        let span = data.get_span(id.index, sess);

        let macro_full_name = data.def_path(id.index).to_string_friendly(|_| data.root.name);
        let source_name = FileName::Custom(macro_full_name);
        let source_content = rustc_ast_pretty::pprust::tts_to_string(body.clone());
        let source_file = sess.parse_sess.source_map().new_source_file(source_name, source_content);
        let span = span.with_lo(source_file.start_pos).with_hi(source_file.end_pos);
```
to see if the primary change was the cause, but it didn't result in reverting this case, so I'm thinking that's not the cause.

So at this point I don't really have a clue what could have caused this diff.